db/document.go

```diff
@@ -190,6 +190,24 @@ func (doc *Document) MarshalBodyAndSync() (retBytes []byte, err error) {
 	}
 }
 
+func (doc *Document) MarshalBodyForWebhook() (retBytes []byte, err error) {
+	bodyBytes, err := doc.BodyBytes()
```

Comment:
With these lazy marshal/unmarshal functions - it's actually incredibly difficult to determine whether this will cause an actual marshal or not.

Might be worth revisiting that idea in the future.

At the very least we could log at trace level when we're doing or not doing a marshal/unmarshal so it's easy to see if we're doing a bunch of extra work in some places.